zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/distribution/CommandDistributionBehavior.java

```diff
@@ -45,20 +60,41 @@ public CommandDistributionBehavior(
   }
 
   /**
-   * Distributes a command to the other partitions
+   * Distributes a command to all other partitions.
    *
-   * @param distributionKey the key which is used for the distribution. This could either be a new
-   *     key, but it could also be the key of the entity that's getting distributed.
-   * @param command the command that needs to be distributed
+   * @param distributionKey the key to identify this unique command distribution. The key is used to
+   *     store the pending distribution, as the key of distributed command, to identify the
+   *     distributing partition when processing the distributed command, and as the key to correlate
+   *     the ACKNOWLEDGE command to the pending distribution. This can be a newly generated key, or
+   *     the key identifying the entity that's being distributed. Please note that it must be unique
+   *     for command distribution. Don't reuse the key to distribute another command.
+   * @param command the command to distribute
    */
   public <T extends UnifiedRecordValue> void distributeCommand(
       final long distributionKey, final TypedRecord<T> command) {
-    if (otherPartitions.isEmpty()) {
+    distributeCommand(distributionKey, command, otherPartitions);
+  }
+
+  /**
+   * Distributes a command to the specified partitions.
+   *
+   * @param distributionKey the key to identify this unique command distribution. The key is used to
+   *     store the pending distribution, as the key of distributed command, to identify the
+   *     distributing partition when processing the distributed command, and as the key to correlate
+   *     the ACKNOWLEDGE command to the pending distribution. This can be a newly generated key, or
+   *     the key identifying the entity that's being distributed. Please note that it must be unique
+   *     for command distribution. Don't reuse the key to distribute another command.
+   * @param command the command to distribute
+   * @param partitions the partitions to distribute the command to
+   */
+  public <T extends UnifiedRecordValue> void distributeCommand(
+      final long distributionKey, final TypedRecord<T> command, final List<Integer> partitions) {
+    if (partitions.isEmpty()) {
       return;
     }
 
     final var distributionRecord =
-        new CommandDistributionRecord()
+        commandDistributionStarted
```

Comment:
💭 What about implementing a reset method for the record and calling it before or after each usage? For now, it is not used around, so seems clear but for future changes, that can make the re-using more explicit. WDYT?